app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SessionsController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      flash[:notice] = "Welcome back"
+      redirect_to root_path
+    else
+      flash[:notice] = "Invalid email or password"
+      render 'new'
+    end
+  end
+
+  def destroy
+    @_current_user = session[:user_id] = nil
```

Comment:
you don't need to negate the `@_current_user` since instance variables only last the length of the request, and you're immediately redirecting.